Others/Sudoku.java

```diff
@@ -0,0 +1,87 @@
+import java.util.Scanner;
+
+class Sudoku {
+
+    public static void main(String[] args) {
+        Scanner scn = new Scanner(System.in);
+
+        int[][] arr = new int[9][9];
+        for (int i = 0; i < arr.length; ++i) {
+            for (int j = 0; j < arr[0].length; ++j) {
+                arr[i][j] = scn.nextInt();
+            }
+        }
+
+        int[] rows = new int[9];
+        int[] cols = new int[9];
+        int[][] sms = new int[3][3];
+
+        for (int row = 0; row < arr.length; ++row) {
+            for (int col = 0; col < arr[0].length; ++col) {
+                rows[row] |= (1 << arr[row][col]);
+                cols[col] |= (1 << arr[row][col]);
+                sms[row / 3][col / 3] |= (1 << arr[row][col]);
+            }
+        }
+
+        sudoku(arr, 1, rows, cols, sms);
+    }
+
+    private static void sudoku(int[][] arr, int cellno, int[] rows, int[] cols, int[][] sms) {
+        if (cellno > arr.length * arr.length) {
+            for (int i = 0; i < arr.length; ++i) {
+                for (int j = 0; j < arr[0].length; ++j) {
+                    System.out.print(arr[i][j] + " ");
+                }
+                System.out.println();
+            }
+        }
+
+        int row = (cellno - 1) / arr.length;
+        int col = (cellno - 1) % arr.length;
+
+        if (arr[row][col] != 0) {
+            sudoku(arr, cellno + 1, rows, cols, sms);
+        } else {
+            for (int choice = 1; choice <= 9; ++choice) {
```

Comment:
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 9 out of bounds for length 9